hnn_core/__init__.py

```diff
@@ -6,5 +6,6 @@
 from .cell_response import CellResponse, read_spikes
 from .cells_default import pyramidal, basket
 from .parallel_backends import MPIBackend, JoblibBackend
+from optimization.optimizer import Optimizer
```

Comment:
```suggestion
from .optimization.optimizer import Optimizer
```

Reply:
To avoid conflict with the previously created "optimization.py" file, we should probably rename it and move it into the "optimization" directory.